addons/l10n_ro/migrations/14.0.2.0/pre-update_account_tax_report_line.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+# Copyright (C) 2021 NextERP Romania (https://www.nexterp.ro) <contact@nexterp.ro>
+
+def migrate(cr, version):
+    cr.execute("UPDATE account_tax_report_line "
+               "SET tag_name = NULL "
+               "WHERE code = 'tax_ro_baza_rd24'")
+    cr.execute("UPDATE account_tax_report_line "
+               "SET tag_name = NULL "
+               "WHERE code = 'tax_ro_baza_rd25'")
+    cr.execute("UPDATE account_tax_report_line "
+               "SET tag_name = NULL "
+               "WHERE code = 'tax_ro_baza_rd26'")
```

Comment:
This would have been correct on any other model, but not here :p 

Tax report lines create account.tag objects depending on their tag_name field. If it's modified, the write() function is overridden to also impact the tags, possibly deleting them (if the tag_name is now empty). https://github.com/odoo/odoo/blob/14.0/addons/account/models/account_tax_report.py#L148

So, in essence, to do this, please use the ORM in post or end instead ;)

Reply:
@oco-odoo I would do it but since the data in xml has changed, when it is updated will raise an error since tag_name and formula are mutually exclusive, that's why i put them in pre.

Reply:
@feketemihai I get your point, but this won't exactly do what you want. 
You don't only want to remove the tag_name from these tax report lines. You also want to delete the tags it was associated with, and to remove them from all the taxes and move lines referencing them. That's why you need to call the ORM to set tag_name to None: the write() is overridden to do all this tags management for you. 

I gave it a second thought, and there is actually a way easier solution, you don't even need the migration script. As you witnessed, just removing <field name="tag_name">...</field> from the xml isn't reassigning the field to None. However, if you simply replace these lines by <field name="tag_name"></field>, it will. This way, the ORM will reset the field as empty (so, None), and remove the tags just as we want it to :)

Reply:
@oco-odoo I fixed this in the last commit